include/f_core/device/sensor/c_sensor_device.h

```diff
@@ -0,0 +1,79 @@
+#ifndef C_SENSOR_DEVICE_H
+#define C_SENSOR_DEVICE_H
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+
+class CSensorDevice {
+public:
+    /**
+     * Constructor
+     * @param[in] device Zephyr device structure
+     */
+    CSensorDevice(const device &device) : dev(device), isInitialized(false) {}
+
+    /**
+     * Set isInitialized to true if the device is ready
+     * @return true if the device is initialized, false otherwise
+     */
+    bool Initialize() {
```

Comment:
Been on the fence about this too. Just been sticking with the years of just doing this (even in OG launch core). The main reason we have Init functions instead of doing things in constructors is so we can propogate a return value up to signal any errors). I think in some cases its fine to have an init function.

Reply:
For this use case, might be fine